src/extension/background-script/actions/nostr/enable.ts

```diff
@@ -25,8 +25,22 @@ const enable = async (message: MessageAllowanceEnable, sender: Sender) => {
 
   const enabledFor = new Set(allowance?.enabledFor);
 
+  if (!isUnlocked) {
+    try {
+      await utils.openPrompt<{ unlocked: boolean }>({
+        args: {},
+        origin: { internal: true },
+        action: "unlock",
+      });
+    } catch (e) {
+      console.error(e);
+      if (e instanceof Error) {
+        return { error: e.message };
+      }
+    }
+  }
+
   if (
-    isUnlocked &&
```

Comment:
don't we still need the unlock check? because if your new check fails for whatever reason then it's not unlocked and this action should not run.